### ISSUE TITLE
Support for XLink Kai 'DDS' integration

### ISSUE THREAD

**Issue by CrunchBite** [PCSX2/pcsx2#3878]
With the recent work and merging of https://github.com/PCSX2/pcsx2/pull/3782 this is the ideal time to add support for directly exchanging LAN traffic with XLink Kai for online play. This can be accomplished with a UDP socket and a small handshake process then PCSX2 will be able to exchange Ethernet frames. 
I added this feature to XLink Kai a couple versions ago and at the same time added support to Dolphin in https://github.com/dolphin-emu/dolphin/pull/8853
I've written a short spec doc on 'DDS' here: https://www.teamxlink.co.uk/wiki/DDS_Protocol

**Comment by CrunchBite** [comment]
I've expanded the DDS documentation a bit. Added the auto-discovery mechanism and added a compatibility matrix.
https://www.teamxlink.co.uk/wiki/DDS_Protocol

**Comment by GovanifY** [comment]
Thank you @CrunchBite ! I'll try to take a look at it soon, work has been hitting me hard last week

**Comment by CrunchBite** [comment]
No worries! Gives me more time to build an adequate test setup :)

**Comment by GovanifY** [comment]
Yes, it is! I and the team have just been quite busy as of late and we had to prioritize big work lately. Hopefully we will be able to work on this soon, otherwise anyone willing to do so in the meantime can always submit a patch, ping me, and I'll review it!
